src/tlv/tlv_library.h

```diff
@@ -0,0 +1,283 @@
+    X(0x77, TAG_C, NULL, ALLOW_MULTIPLE_TAG)
+
+DEFINE_TLV_PARSER(MY_TAGS, my_tlv_parser)
+
+bool my_tlv_use_case(buffer_t payload, uint8_t *my_value_to_receive) {
+    my_tlv_output_t out = {0};
+    if (!parse_tlv_trusted_name(&payload, &out, &out.received_tags)) {
+        return false;
+    }
+    if (!CHECK_RECEIVED_TAGS(out.received_tags, TAG_A, TAG_B)) {
+        return false;
+    }
+    *my_value_to_receive = out.my_value_1;
+    return true;
+}
+
+ * ```
+ */
+
+// ─────────────────────────────────────────────────────────────────────────────
+// TLV library API
+// ─────────────────────────────────────────────────────────────────────────────
+
+// The received TLV data that will be fed as input to each handler
+typedef struct tlv_data_s {
+    // The tag is reminded in case the handler is generic and covers multiple tags
+    TLV_tag_t tag;
+    // The length of the value buffer
+    uint16_t length;
+    // The value buffer
+    const uint8_t *value;
+
+    // In case the handler needs to do some processing on the raw TLV-encoded data
+    // Computing a signature for example
+    const uint8_t *raw;
+    uint16_t raw_size;
+} tlv_data_t;
+
+// The handlers prototype to give to give to DEFINE_TLV_PARSER() through the TAG_LIST parameter
+// The type of tlv_extracted must be the same for every handler of a given TLV use case
+typedef bool(tlv_handler_cb_t)(const tlv_data_t *data, void *tlv_extracted);
+
+// The generated TLV parser will set in this structure the flags of each received tag
+// Actual structure content is private, use check_received_tags or CHECK_RECEIVED_TAGS
+typedef TLV_reception_internal_t TLV_reception_t;
+
+typedef enum tag_unicity_e {
+    // The parser will fail if the associated tag is received more than once
+    ENFORCE_UNIQUE_TAG,
+    // The parser will call the handler every time this tag is received
+    ALLOW_MULTIPLE_TAG,
+} tag_unicity_t;
+
+/**
+ * @brief Creates a parser function for a given TLV use case
+ *
+ * @param TAG_LIST A X-macro of TAG and associated handles of the following format:
+ * ```c
+TAG_LIST(X) \
+    X(<uint32_t>, TAG_NAME, <tlv_handler_cb_t>, <tag_unicity_t>)
+ * ```
+ * @param PARSE_FUNCTION_NAME Name of the function that will parse the TAG_LIST
+ *
+ * Example:
+ * ```c
+ * #define MY_TAGS(X) \
+ *     X(0x0A, TAG_A, my_handler_1, ALLOW_MULTIPLE_TAG) \
+ *     X(0x1F, TAG_B, my_handler_2, ENFORCE_UNIQUE_TAG) \
+ *     X(0x77, TAG_C, my_handler_2, ENFORCE_UNIQUE_TAG)
+ *
+ * DEFINE_TLV_PARSER(MY_TAGS, my_tlv_parser)
+ * ```
+ *
+ * Expands to:
+ * - `enum { TAG_A = 0x0A, TAG_B = 0x1F, TAG_C = 0x77};`
+ * - `enum { TAG_A_INDEX, TAG_B_INDEX, TAG_C_INDEX, my_tlv_parser_TAG_COUNT };`
+ * - `enum { TAG_A_FLAG = 1 << 0, TAG_B_FLAG = 1 << 1, TAG_C_FLAG = 1 << 2 };`
+ * - `my_tlv_parser_tag_to_flag()`
+ * - `my_tlv_parser()`
+ */
+// clang-format off
+#define DEFINE_TLV_PARSER(TAG_LIST, PARSE_FUNCTION_NAME)                                 \
```

Comment:
Can't be used by the Rust SDK as the binder does not expand C macros.

Reply:
Handled through use case API